### ISSUE TITLE
Incorrect success reported due to unsupported constructs

### ISSUE THREAD

**Issue by zhassan-aws** [model-checking/rmc#725]
I tried this code:

```rust
#![feature(asm)]
fn foo(x: &mut u8, y: &mut u8) {
    unsafe {
        std::arch::asm!("nop");
    }
    *x = 0;
}
 fn main() {
    let mut x: u8 = rmc::nondet();
    let mut y = x;
    if x > 100 {
        foo(&mut x, &mut y);
    } else {
        x /= 2;
        y /= 2;
    }
    assert!(x == y);
}
```

using the following command line invocation:

```
rmc test.rs
```

with RMC version: `ab096b6ce6a`

I expected to see this happen: Since an unsupported construct (inline assembly) is reachable, I expected RMC to not report results for other assertions since their results are questionable (and in the example above is actually wrong).

Instead, this happened: RMC reported SUCCESS for `assert!(x == y)`:
```
/home/ubuntu/examples/unsupported/test.rs function foo
[foo.assertion.1] line 4 InlineAsm is not currently supported by RMC. Please post your example at https://github.com/model-checking/rmc/issues/2 : FAILURE

/home/ubuntu/examples/unsupported/test.rs function main
[main.assertion.1] line 17 assertion failed: x == y: SUCCESS

** 1 of 10 failed (2 iterations)
VERIFICATION FAILED
```
